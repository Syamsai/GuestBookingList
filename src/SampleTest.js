import React from "react";

const SampleTest = ({ listOfStrings }) => {
 return (
    <ul>
        { listOfStrings?.length > 0 
        ? listOfStrings.map((item, index) => <li key={index}>{item}</li>)
        : <li>NoStrings</li>}
    </ul>
 )
}

export default SampleTest;