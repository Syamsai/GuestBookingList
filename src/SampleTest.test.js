import React from "react";
import { shallow } from "enzyme";   
import SampleTest from "./SampleTest";

describe("Sample Test Component", () => {
    it("Render NoStrings Text", () => {
      const component = shallow(<SampleTest listOfStrings={[]} />)
      expect(component).toEqual({});
    })

    it("Render the two strings", () => {
        const strings = ["three", "four", "five", "six"];
        const component = shallow(<SampleTest listOfStrings={strings}/>)
        expect(component).toMatchSnapshot();
    })
})

const myBeverage = {
    delicious: true,
    sour: false,
  };
  
  describe('my beverage', () => {
    test('is delicious', () => {
      expect(myBeverage.delicious).toBeTruthy();
    });
  
    test('is not sour', () => {
      expect(myBeverage.sour).toBeFalsy();
    });
  });