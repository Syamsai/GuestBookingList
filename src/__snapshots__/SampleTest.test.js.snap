// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Test Component Render the two strings 1`] = `
<ul>
  <li
    key="0"
  >
    three
  </li>
  <li
    key="1"
  >
    four
  </li>
  <li
    key="2"
  >
    five
  </li>
  <li
    key="3"
  >
    six
  </li>
</ul>
`;
